imod/protocols/protocol_base.py

```diff
@@ -417,75 +417,59 @@ def addCTFTomoSeriesToSetFromDefocusFile(self, inputTs, defocusFilePath, output)
 
         else:
             raise Exception(
-                "Defocus file flag do not supported. Only supported formats "
-                "corresponding to flags 0, 1, 4, 5, and 37.")
+                f"Defocus file flag {defocusFileFlag} is not supported. Only supported formats "
+                "correspond to flags 0, 1, 4, 5, and 37.")
 
+        excludedViews = inputTs.getExcludedViewsIndex()
         for index, _ in enumerate(inputTs):
```

Comment:
This , although will work is a typical mistake (it was there before)

enumerating a set is a costly effort, and in this case is even worse cause we just want the index! and later we do the 

inputTS[index- +1]

So I'd suggest to either:

use the item in the enumerate or use set.getIds() and leave the inputTS[index- +1]

Reply:
got it. Found this in 11 files, will fix this now.